### DISCUSSION TITLE
Welcome to .config Discussions!

### DISCUSSION THREAD

**Discussion by benbrastmckie** [benbrastmckie/.config#103] [Announcements]
I've opened this discussion for thoughts and other considerations regarding the existing state of the config, additional plugins or directions worth considering, feedback on how the config is working for you, what you've done to extend it, and anything else that doesn't belong in the issues. Happy Viming!

**Comment by aidangotgit** [comment]
Hello! Really loving this .config, thanks for putting all this effort in to create and share it. I am not too well versed in nvim plugins and was wondering how you would recommend installing extra plugins in a way that keeps with the current config organization. Thanks again!

**Comment by benbrastmckie** [comment]
Hey good question! It's getting easier all the time. If you are familiar with making changes to your config, then go check out some of the files in `~/.config/nvim/lua/neotex/plugins/`. If you are not so familiar, you can easily navigate there from the home screen with `c` and then `<space>e`, navigating to the directory just mentioned.

You will notice that they all begin with `return` and a lua table, i.e., the brackets following `return`, and then list the plugin name in quotes, various dependencies if any, and configuration if any. All you need to do is go to the github repo for the plugin you want to add, look at installation for lazy.nvim (they typically have this now, at least for current plugins), and reproduce something similar to the following:

```
return {
    "author/plugin-name",
}
```

In the easiest case, that's all you have to do. Save the file and relaunch neovim. If you want to add dependencies and configuration, then you can cut and past whatever appeals from the plugin github repo as follows:

```
return {
    "author/plugin-name",
    dependencies = { ADD ANY RECOMMENDED ON THE PLUGIN GITHUB },
    config = function()
        OPTIONAL CONFIG STUFF YOU CAN CUT AND PASTE FROM THE GITHUB
    end
}
```

Out of curiosity, what plugins were you thinking of adding? Also, if you have encountered any errors in the documentation, you should be able to edit the docs on my repo and submit as a pull request. But no worries if you haven't kept track.

**Reply by aidangotgit** [reply]
This is fantastic, thank you! The two plugins are [hardtime](https://github.com/m4xshen/hardtime.nvim) because I am stubborn and new-ish to vim and [zen-mode](https://github.com/folke/zen-mode.nvim). The latter of which (unless I am missing it in the current setup) should probably be included in a writing focused config like this. Most people are used to the skinnier layout from Google Docs/Microsoft Word for one thing. And I think here specifically turning off spell check, auto complete, etc. when entering the mode could be a fitting implementation. I would offer to set that up myself but I don't quite trust myself and midterm season is upon me :)

**Comment by benbrastmckie** [comment]
I know about `zen-mode` but haven't seen `hardtime`. I don't find zen-mode helpful personally, but have added it to the `~/.config/nvim/lua/neotex/deprecated/` directory for people to swap in as they please. I will also include a key binding in `whichkey.lua` that is commented out. I haven't added no spell check or autocomplete but it shouldn't be hard. See the end of the `zen-mode.lua` file.

Perhaps also worth mentioning: none of the files in `/deprecated/` are loaded, but if you move them into the `/plugins/` directory they will be. This can be done easily with `x` and `p` to cut and past respectively in the file tree. I'll give `hardtime` a try to check it out, and it will probably end up in `deprecated` as well. For now it is the `/plugins/` directory. Hope that helps!

**Comment by cosmiccoder01** [comment]
This is excellent, thank you so much for putting this together. 

I am facing one issue with getting citations work. I use this command `\cite{key}` It says compilation failed, and the log file says the key is undefined. 

I definitely see this key in the `﻿~/Library/texmf/bibtex/bib/Zotero`, and in my TeX file I have the following 

```
\bibliographystyle{plain}
\bibliography{Zotero}
```
`<space>fc` gives me all the citations which are in the Zotero.bib file. 

`:checkhealth` gives me no errors
```
vimtex:                                                    health#vimtex#check

VimTeX ~
- OK Vim version should have full support!
- OK Skim viewer should work!
- OK Compiler should work!
```


Is there something I am missing?

**Reply by vasi786** [reply]
Figured it out. Some other errors in my latex file are causing citations to not work.
